crates/external_services/src/crm.rs

```diff
@@ -0,0 +1,157 @@
+use std::{
+    fmt::{Display, Formatter},
+    sync::Arc,
+};
+
+use common_utils::{
+    errors::CustomResult,
+    ext_traits::ConfigExt,
+    request::{Method, Request, RequestBuilder, RequestContent},
+};
+use error_stack::ResultExt;
+use http::header;
+use hyperswitch_interfaces::{
+    crm::{CRMInterface, CRMPayload},
+    errors::HttpClientError,
+    types::Proxy,
+};
+use masking::PeekInterface;
+use reqwest;
+use router_env::logger;
+
+use crate::{http_client, hubspot_proxy::HubspotRequest};
+
+/// Hubspot CRM configuration
+#[derive(Debug, Clone, serde::Deserialize)]
+pub struct HubspotProxyConfig {
+    /// The ID of the Hubspot form to be submitted.
+    pub form_id: String,
+
+    /// The URL to which the Hubspot form data will be sent.
+    pub request_url: String,
+}
+
+impl HubspotProxyConfig {
+    /// Validates Hubspot configuration
+    pub(super) fn validate(&self) -> Result<(), InvalidCRMConfig> {
+        use common_utils::fp_utils::when;
+
+        when(self.request_url.is_default_or_empty(), || {
+            Err(InvalidCRMConfig("request url must not be empty"))
+        })?;
+
+        when(self.form_id.is_default_or_empty(), || {
+            Err(InvalidCRMConfig("form_id must not be empty"))
+        })
+    }
+}
+
+/// Error thrown when the crm config is invalid
+#[derive(Debug, Clone)]
+pub struct InvalidCRMConfig(pub &'static str);
```

Comment:
```suggestion
pub struct InvalidCrmConfig(pub &'static str);
```
Lets not use full CAPITAL in snake-case, do the same in other places too